homeassistant/components/mqtt/lock.py

```diff
@@ -148,8 +157,24 @@ def message_received(msg: ReceiveMessage) -> None:
             payload = self._value_template(msg.payload)
             if payload == self._config[CONF_STATE_LOCKED]:
                 self._attr_is_locked = True
+                self._attr_is_locking = False
```

Comment:
May be we can refactor this a bit by checking the state is in a list of valid states, then assigning with some logic.

LOCK_STATES = [